Tests/IntegrationTests/PostgresNIOTests.swift

```diff
@@ -341,6 +341,85 @@ final class PostgresNIOTests: XCTestCase {
         XCTAssertEqual(UUID(uuidString: row?[data: "id"].string ?? ""), UUID(uuidString: "123E4567-E89B-12D3-A456-426655440000"))
     }
 
+    func testInt4Range() throws {
```

Comment:
we should add tests with which we also test the encoding...

Reply:
Added to `Tests/PostgresNIOTests/New/Data/Range+PSQLCodableTests.swift` in https://github.com/vapor/postgres-nio/pull/330/commits/26b59f03b0db71fabc822bc0ee33e8d20c09b00d.

Reply:
Added integration tests here https://github.com/vapor/postgres-nio/pull/330/commits/42174af06fcf7af5547dc08f990e288490fc0460.

I started to convert tests to async, but I wasn't sure how to handle the many instances of:
```
defer { XCTAssertNoThrow( try conn?.close().wait() ) }
```

If I use `await` instead of `wait()`, then I can't use `XCTAssertNoThrow`. If I don't use `XCTAssertNoThrow`, then the `defer` block won't build due to `errors cannot be thrown out of a defer body`. If I catch the error, I still get `'async' call cannot occur in a defer body`.

Any thoughts about how you'd like that handled?

Reply:
Thanks, done in https://github.com/vapor/postgres-nio/pull/330/commits/618a9d700c5b97ab247671e4467aabc41b3299b9.